api/routes/image_pack.go

```diff
@@ -0,0 +1,197 @@
+			IDs = append(IDs, r.IDs...)
+		}
+		err = handleJSON(w, ImagePackResult{
+			ImagesBuffer: imagesBuffer,
+			ImageIDs:     IDs,
+		})
+		if err != nil {
+			handleError(w, errors.Wrap(err, "unable marshal dataset result into JSON"))
+			return
+		}
+	}
+}
+func getImages(imagePackRequest *ImagePackRequest, threadID int, numThreads int, result chan chanStruct, ctor api.MetadataStorageCtor, dataCtor api.DataStorageCtor) {
+	temp := [][]byte{}
+	IDs := []string{}
+	for i := threadID; i < len(imagePackRequest.ImageIDs); i += numThreads {
+		imageID := imagePackRequest.ImageIDs[i]
+		// get metadata client
+		storage, err := ctor()
+		if err != nil {
+			continue
+		}
+
+		res, err := storage.FetchDataset(imagePackRequest.Dataset, false, false, false)
+		if err != nil {
+			continue
+		}
+
+		sourcePath := env.ResolvePath(res.Source, res.Folder)
+
+		data, err := ioutil.ReadFile(path.Join(sourcePath, imageFolder, imageID))
+		if err != nil {
+			continue
+		}
+		img, _, err := image.Decode(bytes.NewReader(data))
+		if err != nil {
+			continue
+		}
+		img = resize.Resize(ThumbnailDimensions, ThumbnailDimensions, img, resize.Lanczos3)
+		rgbaImg := image.NewRGBA(image.Rect(0, 0, ThumbnailDimensions, ThumbnailDimensions))
+		draw.Draw(rgbaImg, image.Rect(0, 0, ThumbnailDimensions, ThumbnailDimensions), img, img.Bounds().Min, draw.Src)
+		imageBytes, err := util.ImageToJPEG(rgbaImg)
+		if err != nil {
+			continue
+		}
+		temp = append(temp, imageBytes)
+		IDs = append(IDs, imageID)
+	}
+	result <- chanStruct{data: temp, IDs: IDs}
+}
+func getMultiBandImages(multiBandPackRequest *ImagePackRequest, threadID int, numThreads int, result chan chanStruct, ctor api.MetadataStorageCtor, dataCtor api.DataStorageCtor) {
+	temp := [][]byte{}
+	IDs := []string{}
+	for i := threadID; i < len(multiBandPackRequest.ImageIDs); i += numThreads {
+		imageID := multiBandPackRequest.ImageIDs[i]
+		// get metadata client
+		storage, err := ctor()
+		if err != nil {
+			continue
+		}
+		dataStorage, err := dataCtor()
+		if err != nil {
+			continue
+		}
+
+		res, err := storage.FetchDataset(multiBandPackRequest.Dataset, false, false, false)
+		if err != nil {
+			continue
+		}
+		sourcePath := env.ResolvePath(res.Source, res.Folder)
+
+		// need to read the dataset doc to determine the path to the data resource
+		metaDisk, err := metadata.LoadMetadataFromOriginalSchema(path.Join(sourcePath, compute.D3MDataSchema), false)
+		if err != nil {
+			continue
+		}
+		for _, dr := range metaDisk.DataResources {
+			if dr.IsCollection && dr.ResType == model.ResTypeImage {
+				sourcePath = model.GetResourcePathFromFolder(sourcePath, dr)
+				break
+			}
+		}
+		options := util.Options{Gain: 2.5, Gamma: 2.2, GainL: 1.0, Scale: 0} // default options for color correction
+
+		imageScale := util.ImageScale{Width: ThumbnailDimensions, Height: ThumbnailDimensions}
+
+		// need to get the band -> filename from the data
+		bandMapping, err := getBandMapping(res, imageID, dataStorage)
```

Comment:
I think everything up to here is common across all images and should be done outside of the loop